### ISSUE TITLE
Ability to suppress diagnostics

### ISSUE THREAD

**Issue by RonRahaman** [enrico-dev/enrico#89]
Should we implement a way to suppress diagnostic messages?  Currently, this includes the sanity check for volume calculations.  For the disjoint communicators, I also have some diagnostics about the rank layout that I'd like to keep.  But these both get pretty verbose in large-ish runs.

I'm considering either:
* A `-v` runtime flag
* The `NDEBUG` macro.  This is defined by CMake for release builds.  It's also understood by `assert()` as specified by the C++ standard.  However, it doesn't appear to affect the behavior of `Ensures` or `Expects` in GSL.  

I'm tending towards a `-v` flag.

**Comment by paulromano** [comment]
Yes, I agree that a `-v` flag is probably the simplest way to do this.
